src/rgw/rgw_rest_conn.cc

```diff
@@ -438,7 +438,7 @@ void RGWRESTSendResource::init_common(param_vec_t *extra_headers)
 int RGWRESTSendResource::send(bufferlist& outbl)
 {
   req.set_outbl(outbl);
-  int ret = req.send_request(&conn->get_key(), headers, resource, mgr);
+  int ret = req.send_request(&conn->get_key(), headers, resource, mgr, req.get_outbl());
```

Comment:
this call is weird. req already holds the outbl, so maybe find a different way to telll it to set the send size?

Reply:
instead of call set_outbl here, maybe we can specify the `send_data` in req.send_request() and use `set_outbl(*send_data)` in RGWRESTStreamRWRequest::send_prepare, does this in a right way ?

Reply:
@Leeshine I'm not 100% sure, need to take a closer look at everything. One thing to keep in mind is that `RGWRESTStreamRWRequest` can be used without outbl when streaming data. There are also a few subclasses that inherit from it, so I'd be careful not to break these also.

Reply:
@yehudasa if the `send_data` is not specified here, we also need to use `get_outbl()` in `RGWRESTStreamRWRequest` to decide whether to set send size or not,  and in `RGWRESTStreamRWRequest::send_prepare()` , if the `send_data` is specified, it will call `set_outbl()` too,  it seems to be a superfluous call, so maybe we only need to remove `req.set_outbl(outbl)` and specified `send_data` here ? And it should not take some side effects to the subclassed of  `RGWRESTStreamRWRequest` or used when streaming data.

Reply:
@Leeshine can you just call `req.set_send_length(oubl.length())` here? Another option is at `req.set_outbl()` to do a `set_send_length()`. I'm not extremely happy with any of the solutions, there isn't a clear distinction in the interfaces between an actual streaming write and a single write where we have all the data in advance.